lld/COFF/Driver.cpp

```diff
@@ -657,7 +656,7 @@ void LinkerDriver::setMachine(MachineTypes machine) {
 
   ctx.config.machine = machine;
 
-  if (machine != ARM64X) {
+  if (!isArm64EC(machine)) {
```

Comment:
So this change makes us create two symbol tables for all arm64ec linking, even if the very fast majority of them ever only will use one of them?

That feels odd, but I guess it's good for consistency - otherwise we'd probably have lots of bugs for the very rare cases when we do need both symbol tables for arm64ec. And I guess this is the change which forces the extra `(EC symbol)` to be printed in many tests.

Reply:
Yes, that’s right, this is the core part of the change; the rest of the patch updates various components to handle it properly. The extra `(EC symbol)` annotation doesn’t add much value when there’s only a single symbol table, but it becomes quite useful with this PR. If someone accidentally passes a native object file, error messages that include `(native symbol)` should make the issue easy to diagnose.